File: Cousins in Binary Tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    
    bool isCousins(TreeNode* root, int x, int y) {
        queue<pair<TreeNode*,int>>q;
        pair<TreeNode*,int>k,l;
        k.first=root;
        k.second=0;
        q.push(k);
        int depth_x,depth_y;
        int count=0;
        
        while(!q.empty() && count!=2)
        {
            k=q.front();
            q.pop();
            TreeNode* n=k.first;
            int depth=k.second;
            if(n->left!= NULL && n->right!= NULL)
            {
                if((n->left->val==x && n->right->val==y) ||(n->left->val==y && n->right->val==x))
                    return false;
            }
            if(n->val==x)
            {
                depth_x=depth;
                count++;
            }
            else if(n->val==y)
            {
                depth_y=depth;
                count++;
            }
            if(n->left != NULL)
            {
                k.first=n->left;
                k.second=depth+1;
                q.push(k);
                
            }
            if(n->right != NULL)
            {
                k.first=n->right;
                k.second=depth+1;
                q.push(k);
            }
        }
        
        return depth_x==depth_y?true:false;
            
    }
};

File: Golang Solutions/Cousins in Binary Tree.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func isCousins(root *TreeNode, x int, y int) bool {
    q:=[]*TreeNode{}
    q_val:=[]int{}
    q=append(q,root)
    q_val=append(q_val,0)
    var depth_x,depth_y int
    cnt,d:=0,0
    for ;len(q)!=0 && cnt<=2; {
        var n *TreeNode= new(TreeNode)
        n=q[0]
        d=q_val[0]
        q=q[1:]
        q_val=q_val[1:]
        if(n.Left != nil && n.Right != nil){
            if((n.Left.Val==x && n.Right.Val==y) ||(n.Left.Val==y && n.Right.Val==x)){
                return false;
            }         
        }
        if n.Val==x{
            depth_x=d
            cnt++
        }else if(n.Val == y){
            depth_y=d
            cnt++
        }
        if(n.Left != nil){
            q=append(q,n.Left)
            q_val=append(q_val,d+1)  
        }
        if(n.Right != nil){
             q=append(q,n.Right)
            q_val=append(q_val,d+1) 
        }
        
    }
    if depth_x==depth_y{
        return true
    }
    return false      
}